main.tf

```diff
@@ -38,8 +38,22 @@ locals {
   nat_instance_eip_ids    = local.reuse_nat_instance_eips ? var.nat_instance_eip_ids : aws_eip.nat_instance_eips[*].id
 }
 
+
+
+resource "aws_eip" "protected_nat_instance_eips" {
+  count = local.reuse_nat_instance_eips ? 0 : var.prevent_destroy_eips ? length(var.vpc_az_maps) : 0
+
+  tags = merge(var.tags, {
+    "Name" = "alternat-instance-${count.index}"
+  })
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
 resource "aws_eip" "nat_instance_eips" {
-  count = local.reuse_nat_instance_eips ? 0 : length(var.vpc_az_maps)
+  count = local.reuse_nat_instance_eips ? 0 : var.prevent_destroy_eips ? 0 : length(var.vpc_az_maps)
```

Comment:
```suggestion
count = local.reuse_nat_instance_eips
  ? 0
  : (var.prevent_destroy_eips ? 0 : length(var.vpc_az_maps))
```

Reply:
This syntax isn't allowed:

│ Error: Argument or block definition required

We have to use ()